astropy/units/quantity.py

```diff
@@ -385,29 +385,31 @@ def __array_prepare__(self, obj, context=None):
                                         args[0].__class__.__name__,
                                         args[1].__class__.__name__))
 
-        # In the case of np.power, the unit itself needs to be modified by an
-        # amount that depends on one of the input values, so we need to treat
-        # this as a special case.
-        # TODO: find a better way to deal with this case
-        if function is np.power and result_unit != dimensionless_unscaled:
-
-            if units[1] is None:
-                p = args[1]
+        # In the case of np.power and np.float_power, the unit itself needs to
+        # be modified by an amount that depends on one of the input values,
+        # so we need to treat this as a special case.
+        # TODO: find a better way to deal with this.
+        if result_unit is False:
```

Comment:
`is False` is somewhat problematic because the singleton-property of True and False is an implementation detail of CPython and may change in future versions (unlikely) and may not work on other implementations like PyPy.